docs/metrics-reporting.md

```diff
@@ -0,0 +1,31 @@
+# Set up metrics reporting in custom module
+1. Add the following dependencies to your Cargo.toml
+```[dependencies]
+cb-metrics.workspace = true
+prometheus.workspace = true
+lazy_static.workspace = true```
+2. Import the necessary dependencies:
+```use cb_metrics::sdk::MetricsProvider;
+use lazy_static::lazy_static;
+use prometheus::{IntCounter, Registry};```
+3. Lazy load the prometheus registry and custom metric (for example a counter):
+```// You can define custom metrics and a custom registry for the business logic of your module. These
+// will be automatically scaped by the Prometheus server
+lazy_static! {
+    pub static ref MY_CUSTOM_REGISTRY: prometheus::Registry =
+        Registry::new_custom(Some("da_commit".to_string()), None).unwrap();
+    pub static ref SIG_RECEIVED_COUNTER: IntCounter =
+        IntCounter::new("signature_received", "successful signatures requests received").unwrap();
+}```
+4. Initialize the registry and run the Module's metrics reporting server in main:
+```#[tokio::main]
```

Comment:
this is not formatted properly.

Make sure that it is rendered properly and you can include the ```rust tag so that there is syntax highlights